### PR TITLE
Optimize eth_call

### PR DESCRIPTION
**Pull request by LeoGuo621**
Add MarshalJSON/UnmarshalJSON meth

- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [x] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/types/params.go [hunk 1]
```diff
@@ -77,6 +77,49 @@ func (p Params) String() string {
...
+func (p *Params) UnmarshalJSON(input []byte) error {
+	alias := &struct {
+		EnableCreate                      bool   `json:"enable_create"`
+		EnableCall                        bool   `json:"enable_call"`
+		ExtraEIPs                         []int  `json:"extra_eips"`
+		EnableContractDeploymentWhitelist bool   `json:"enable_contract_deployment_whitelist"`
+		EnableContractBlockedList         bool   `json:"enable_contract_blocked_list"`
+		MaxGasLimitPerTx                  uint64 `json:"max_gas_limit_per_tx"`
+	}{}
+	if err := json.Unmarshal(input, &alias); err != nil {
```
Comment (BananaLF) [inline x/evm/types/params.go]: alias is already point why put point of point in this line

Comment (LeoGuo621) [reply x/evm/types/params.go]: fixed

**Review by BananaLF — DISMISSED** [review]

File: x/evm/watcher/watcher.go [hunk 2]
```diff
@@ -412,6 +413,12 @@ func (w *Watcher) commitBatch(batch []WatchMessage) {
 			w.store.Delete(key)
 		} else {
 			w.store.Set(key, value)
+			//need update params
+			if bytes.Compare(key, prefixParams) == 0 {
```
Comment (giskook) [inline x/evm/watcher/watcher.go]: Maybe we can expand a type for params, so we do not need to always compare bytes here.

Comment (LeoGuo621) [reply x/evm/watcher/watcher.go]: down

**Review by BananaLF — DISMISSED** [review]

**Review by xiangjianmeng — APPROVED** [review]

**Pull request merged**